client/src/main/scala/com/dwolla/cloudflare/domain/model/ratelimits/package.scala

```diff
@@ -0,0 +1,173 @@
+package com.dwolla.cloudflare.domain.model
+
+import com.dwolla.circe._
+import io.circe.{Decoder, Encoder}
+import shapeless.tag.@@
+
+package object ratelimits {
+
+  type RateLimitId = String @@ RateLimitIdTag
+
+  private[cloudflare] val tagRateLimitId: String ⇒ RateLimitId = shapeless.tag[RateLimitIdTag][String]
+
+}
+
+package ratelimits {
+  import java.time.Duration
+
+  import io.circe.ObjectEncoder
+  import io.circe.export.Exported
+  import io.circe.generic.auto._
+
+  trait RateLimitIdTag
+
+  case class RateLimit(id: Option[RateLimitId] = None,
+                       disabled: Option[Boolean] = None,
+                       description: Option[String] = None,
+                       `match`: RateLimitMatch,
+                       bypass: Option[Seq[RateLimitBypass]] = None,
+                       threshold: Int,
+                       period: Duration,
+                       action: RateLimitAction)
+
+  case class RateLimitMatch(request: RateLimitMatchRequest, response: Option[RateLimitMatchResponse] = None)
+
+  case class RateLimitMatchRequest(methods: Option[Seq[Method]] = None,
+                                   schemes: Option[Seq[Scheme]] = None,
+                                   url: String)
+
+  sealed trait Method
+  object Method {
+    case object Get extends Method
+    case object Post extends Method
+    case object Put extends Method
+    case object Delete extends Method
+    case object Patch extends Method
+    case object Head extends Method
+    case object All extends Method
+  }
+
+  sealed trait Scheme
+  object Scheme {
+    case object Http extends Scheme
+    case object Https extends Scheme
+    case object All extends Scheme
+  }
+
+  case class RateLimitMatchResponse(origin_traffic: Option[Boolean] = None,
+                                    headers: Seq[RateLimitMatchResponseHeader])
+
+  case class RateLimitMatchResponseHeader(name: String,
+                                          op: Op,
+                                          value: String)
+
+  sealed trait Op
```

Comment:
What's an `Op`?

Reply:
It's short for `Operator`:

<img width="778" alt="Screen Shot 2019-09-06 at 9 53 12 AM" src="https://user-images.githubusercontent.com/597530/64437627-3aae2180-d08c-11e9-8908-14566beaa6b4.png">